spring-integration-jdbc/src/test/java/org/springframework/integration/jdbc/lock/JdbcLockRegistryTests-context.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+	   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	   xmlns:jdbc="http://www.springframework.org/schema/jdbc"
+	   xmlns:context="http://www.springframework.org/schema/context"
+	   xsi:schemaLocation="http://www.springframework.org/schema/jdbc http://www.springframework.org/schema/jdbc/spring-jdbc.xsd
+			http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+			http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
+
+	<jdbc:embedded-database id="dataSource" type="DERBY"/>
+
+	<jdbc:initialize-database data-source="dataSource" ignore-failures="ALL">
+		<jdbc:script location="${int.drop.script}"/>
+		<jdbc:script location="${int.schema.script}"/>
+	</jdbc:initialize-database>
+
+	<context:property-placeholder location="int-${ENVIRONMENT:derby}.properties"
+								  system-properties-mode="OVERRIDE"
+								  ignore-unresolvable="true"
+								  order="1"/>
+
+	<bean id="transactionManager" class="org.springframework.jdbc.datasource.DataSourceTransactionManager">

```

Comment:
Just have added this : 
```
<tx:annotation-driven/>
```
and ended up with : 
```
 Error creating bean with name 'lockRegistry' defined in class path resource [org/springframework/integration/jdbc/lock/JdbcLockRegistryTests-context.xml]: 
Unsatisfied dependency expressed through constructor parameter 0: Could not convert argument value of type [com.sun.proxy.$Proxy21] to required type [org.springframework.integration.jdbc.lock.JdbcClient]: 
```
And that is logical since `JdbcClient` is `@Transactional`. That's why I said before that `TransactionTemplate` would be better.

`proxy-target-class="true"` helps though... But should not be panacea...